docs/OperatorBestPractices.md

```diff
@@ -0,0 +1,103 @@
+# Operator Best Practices
+
+## Introduction
+
+The goals of the Service Binding Operator is to make it easier for
+applications developers to bind applications with needed backing
+services, without having to perform manual configuration of secrets,
+configmaps, etc. and to assist operator providers in promoting and
+expanding the adoption of their operators. 
+
+When a ServiceBindingRequest is created the Service Binding Operator
+collects binding information and shares it with application. The
+Binding Service Operator's controller injects the binding information
+into the application's "DeploymentConfig", "Deployment" or “Replicaset"
+as environment variables via an intermediate Secret called "binding-request".
+The binding also works with Knative services as it works with any "deployment"
+controller which has the podspec defined in the its jsonpath as
+"spec.template.spec.containers".
+
+This document provides "best practices" guidelines for the development of
+Operators that manage backing services to be bound together with applications
+by the Service Binding Operator.
+
+## Making an Operator Managed Backing Service Bindable
+
+In order to make a service bindable, the operator provider needs to express
+the information needed by applications to bind with the services provided by
+the operator. In other words, the operator provider must express the
+information that is “interesting” to applications.
+
+There are three methods for making Operator Managed Backing Service Bindable:
+
+* [Operator Providing Metadata in OLM](#operator-providing-metadata-in-olm)
+* [Operator Providing Metadata not in OLM](#operator-providing-metadata-not-in-olm)
+* [Operator Not Providing Metadata](#operator-not-providing-metadata)
+
+### Operator Providing Metadata in OLM
+
+This feature enables operator providers to specify binding information an
+operator's OLM (Operator Lifecycle Manager) descriptor. The Service Binding
+Operator extracts to bind the application together with the backing service.
+The information may be specified in the "status" and/or "spec" section of the
+OLM in plaintext or as a reference to a secret.
+
+For example, this is an example of the *bind-able* operator OLM Descriptor.
+In this case for a PostgreSQL database backing operator. 
+``` yaml
+---
+[...]
+statusDescriptors:
+  description: Name of the Secret to hold the DB user and password
+    displayName: DB Password Credentials
+    path: dbCredentials
+    x-descriptors:
+      - urn:alm:descriptor:io.kubernetes:Secret
+      - binding:env:object:secret:user
+      - binding:env:object:secret:password
+  description: Database connection IP address
+    displayName: DB IP address
+    path: dbConnectionIP
+    x-descriptors:
+      - binding:env:attribute
+```
+
+The operator is available [here.](https://github.com/operator-backing-service-samples/postgresql-operator)
+
+### Operator Providing Metadata not in OLM
```

Comment:
This section already looks good! Mark this as something like "Coming soon!" and I am happy to take this in.

Reply:
Done!